From 122342cef92eb9ee5bda22c63839abf9f58a1650 Mon Sep 17 00:00:00 2001
From: Tingguo Cheng <quic_tingguoc@quicinc.com>
Date: Sun, 29 Sep 2024 12:50:16 +0800
Subject: [PATCH 22/68] arm64: dts: qcom: qcs615: Adds SPMI bus, PMIC and
 peripherals

Create a new file qcs615-pmic.dtsi to add on-board PMIC and peripher-
als in the PMIC that controlled by SPMI bus. The peripherals include
GPIO, RTC, PON, power key and resin-key for QCS615 platform. The power
key and volume down key are controlled by PMIC PON hardware on QCS615.

Signed-off-by: Tingguo Cheng <quic_tingguoc@quicinc.com>
---
 arch/arm64/boot/dts/qcom/qcs615-pmic.dtsi | 56 +++++++++++++++++++++++
 arch/arm64/boot/dts/qcom/qcs615-ride.dts  |  1 +
 arch/arm64/boot/dts/qcom/qcs615.dtsi      | 29 +++++++++++-
 3 files changed, 84 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/qcom/qcs615-pmic.dtsi

diff --git a/arch/arm64/boot/dts/qcom/qcs615-pmic.dtsi b/arch/arm64/boot/dts/qcom/qcs615-pmic.dtsi
new file mode 100644
index 000000000000..d5e98503bb32
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/qcs615-pmic.dtsi
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ */
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/spmi/spmi.h>
+
+&spmi_bus {
+	pmm6155au_0: pmic@0 {
+		compatible = "qcom,spmi-pmic";
+		reg = <0x0 SPMI_USID>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pmm6155au_0_pon: pon@800 {
+			compatible = "qcom,pm8998-pon";
+			reg = <0x800>;
+
+			pon_pwrkey: pwrkey {
+				compatible = "qcom,pm8941-pwrkey";
+				interrupts = <0x0 0x8 0x0 IRQ_TYPE_EDGE_BOTH>;
+				debounce = <15625>;
+				bias-pull-up;
+				linux,code = <KEY_POWER>;
+			};
+
+			pon_resin: resin {
+				compatible = "qcom,pm8941-resin";
+				interrupts = <0x0 0x8 0x1 IRQ_TYPE_EDGE_BOTH>;
+				debounce = <15625>;
+				bias-pull-up;
+				linux,code = <KEY_VOLUMEDOWN>;
+			};
+		};
+
+		pmm6155au_0_rtc: rtc@6100 {
+			compatible = "qcom,pm8941-rtc";
+			reg = <0x6100>, <0x6200>;
+			reg-names = "rtc", "alarm";
+			interrupts = <0x0 0x61 0x1 IRQ_TYPE_EDGE_RISING>;
+			allow-set-time;
+		};
+
+		pmm6155au_0_gpios: gpio@c000 {
+			compatible = "qcom,pm8150-gpio";
+			reg = <0xc000>;
+			gpio-controller;
+			gpio-ranges = <&pmm6155au_0_gpios 0 0 12>;
+			#gpio-cells = <2>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/qcom/qcs615-ride.dts b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
index 4ef969a6af15..b79c22730920 100644
--- a/arch/arm64/boot/dts/qcom/qcs615-ride.dts
+++ b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
@@ -6,6 +6,7 @@
 
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
 #include "qcs615.dtsi"
+#include "qcs615-pmic.dtsi"
 / {
 	model = "Qualcomm Technologies, Inc. QCS615 Ride";
 	compatible = "qcom,qcs615-ride", "qcom,qcs615";
diff --git a/arch/arm64/boot/dts/qcom/qcs615.dtsi b/arch/arm64/boot/dts/qcom/qcs615.dtsi
index b6fdf37364c5..fc279bcdcda4 100644
--- a/arch/arm64/boot/dts/qcom/qcs615.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs615.dtsi
@@ -552,8 +552,33 @@ sram@c3f0000 {
 			reg = <0x0 0x0c3f0000 0x0 0x400>;
                 };
 
-		apps_smmu: iommu@15000000 {
-			compatible = "qcom,qcs615-smmu-500", "qcom,smmu-500", "arm,mmu-500";
+                spmi_bus: qcom,spmi@c440000 {
+                	compatible = "qcom,spmi-pmic-arb";
+                	reg = <0x0 0xc440000 0x0 0x1100>,
+                	      <0x0 0xc600000 0x0 0x2000000>,
+                	      <0x0 0xe600000 0x0 0x100000>,
+                	      <0x0 0xe700000 0x0 0xa0000>,
+                	      <0x0 0xc40a000 0x0 0x26000>;
+                	reg-names = "core",
+                	            "chnls",
+                	            "obsrvr",
+                	            "intr",
+                	            "cnfg";
+                	interrupts-extended = <&pdc 1 IRQ_TYPE_LEVEL_HIGH>;
+                	interrupt-names = "periph_irq";
+                	interrupt-controller;
+                	#interrupt-cells = <4>;
+                	#address-cells = <2>;
+                	#size-cells = <0>;
+                	cell-index = <0>;
+                	qcom,channel = <0>;
+                	qcom,ee = <0>;
+               	};
+
+               	apps_smmu: iommu@15000000 {
+               		compatible = "qcom,qcs615-smmu-500",
+               		             "qcom,smmu-500",
+               		             "arm,mmu-500";
 			reg = <0x0 0x15000000 0x0 0x80000>;
 			#iommu-cells = <2>;
 			#global-interrupts = <1>;
-- 
2.34.1

